src/containers/Tasks.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { addTask, deleteTask, completeTask, editState, editTask } from '../actions/index';
+import '../css/tasks.css';
+
+
+class Tasks extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      editing: false
+    };
+    this.handleEditing = this.handleEditing.bind(this);
+  }
+
+  handleEditing = () => {
+    this.setState({editing: !this.state.editing});
+  }
+
+  edit = (e) => {
+    const str = this.refs.editTask.value;
+    if (/\S/.test(str)) {
+      this.props.editTask(str, e);
+    }
+  }
+
+ add = () => {
+   const str = this.refs.addTask.value;
+   if (/\S/.test(str)) {
+     this.props.addTask(str, this.props.user.id);
+   }
+   this.refs.addTask.value = '';
+ }
+
+  showTasks() {
+    return this.props.tasks.map((task, i) => {
+      if (!task.edit) {
+        return(
+          <li key={i} title='Done?' className='task__wrapper'>
+            <span className='task__number'>{i+1}</span>
+            <span className={task.complete ? 'task__text complete' : 'task__text'}>{task.name}</span>
+              <button
+                onClick={() => { this.handleEditing(); this.props.editState(i); }}
```

Comment:
bad practice, use one method and invoke the following two functions inside it (name it properly)